apps/price_pusher/src/evm/chains.ts

```diff
@@ -0,0 +1,347 @@
+import { defineChain } from "viem";
+import * as viemChains from "viem/chains";
+import { Chain } from "viem/chains";
+
+// Get the chain object from the chainId or return an unknown chain object
+// if the chainId is not found in the list of chains. Normally the unknown
+// chain object should work for most cases, but some networks may have special
+// requirements (like custom fee model, etc) that may require a custom chain.
+export function getChain(chainId: number): Chain {
```

Comment:
have you seen https://viem.sh/docs/utilities/extractChain#extractchain?

Reply:
Thanks for referencing it. I saw it in the code but didn't see the example that showed the `Object.values(chains)` will work and ironically when i used it to iterate upon it gave me error. Now I tried it out and it worked.

The `extractChain` itself is not useful here though, because it uses enum of ids and in our case we want to support the unknown chain config in case it's not there as we do have some chains that are not in that list.